app.js

```diff
@@ -0,0 +1,155 @@
+const express = require("express");
+const cors = require("cors");
+const port = 3000;
+const app = express();
+const bodypaser = require("body-parser")
+app.use(cors());
+app.use(bodypaser.urlencoded({extended: false}));
+app.use(bodypaser.json());
+
+const School = require("./School.js");
+const Student = require("./Student.js");
+
+let mySchool = new School();
+
+
+
+
+app.post("/class", (req, res)=>{
+    
+
+    let name = req.body.name;
+    let teacher = req.body.teacher;
+    let newSubject = mySchool.addClass(name, teacher);
+    //route if class or teacher input does not exists
+    if(!name || !teacher){
```

Comment:
These if statements should be done before you add the class to school, you are adding the class before checking.